File: codewars/6-kyu/tribonacci-sequence/solution.rb
# https://www.codewars.com/kata/556deca17c58da83c00002db

def tribonacci(signature,n)
  n < 3 ? signature[0, n] : (n-3).times.each_with_object(signature) { |num, obj| obj << obj[-3] + obj[-2] + obj[-1] }
end

File: codewars/6-kyu/tribonacci-sequence/solution.py
# https://www.codewars.com/kata/556deca17c58da83c00002db

# solution 1: 3273ms
def tribonacci(signature, n):
    lut = signature
    for i in range(3, n):
        lut.append(lut[i-1] + lut[i-2] + lut[i-3])
    return lut[:n]